articles/libraries/data-structures.md

```diff
@@ -67,14 +66,14 @@ ApplyToEach(
 
 ## Oracles ##
 
-In the phase estimation and amplitude amplification literature the concept of an oracle appears frequently.
+In the [phase estimation](https://en.wikipedia.org/wiki/Quantum_phase_estimation_algorithm) and [amplitude amplification](https://en.wikipedia.org/wiki/Amplitude_amplification) literature the concept of an oracle appears frequently.
 Here the term oracle refers to a blackbox quantum subroutine that acts upon a set of qubits and returns the answer as a phase.
 This subroutine often can be thought of as an input to a quantum algorithm that accepts the oracle, in addition to some other parameters, and applies a series of quantum operations and treating a call to this quantum subroutine as if it were a fundamental gate.
 Obviously, in order to actually implement the larger algorithm a concrete decomposition of the oracle into fundamental gates must be provided but such a decomposition is not needed in order to understand the algorithm that calls the oracle.
 In Q#, this abstraction is represented by using that operations are first-class values, such that operations can be passed to implementations of quantum algorithms in a black-box manner.
 Moreover, user-defined types are used to label the different oracle representations in a type-safe way, making it difficult to accidently conflate different kinds of black box operations.
 
-Such oracles appear in a number of different contexts, including famous examples such as Grover's search and quantum simulation algorithms.
+Such oracles appear in a number of different contexts, including famous examples such as [Grover's search and quantum simulation algorithms](https://en.wikipedia.org/wiki/Grover%27s_algorithm).
```

Comment:
I think only "Grover's search" should be a link, not both of them, since simulation are a separate group of algorithms